backend/inspirehep/orcid/utils.py

```diff
@@ -226,3 +232,37 @@ def distributed_lock(lock_name, expire=10, auto_renewal=True, blocking=False):
 
 class DistributedLockError(Exception):
     pass
+
+
+def update_moved_orcid(old_orcid, new_orcid):
+    author_record = AuthorsRecord.get_record_by_pid_value(old_orcid, "orcid")
+    if new_orcid not in get_value(author_record, "ids.value", []):
+        new_author_ids = [
+            {"schema": "ORCID", "value": new_orcid},
+            *author_record["ids"],
+        ]
+        author_record["ids"] = new_author_ids
+        author_record.update(dict(author_record))
+    user_orcid_account = RemoteAccount.query.filter(
+        RemoteAccount.user_id == UserIdentity.id_user, UserIdentity.id == old_orcid
+    ).one()
```

Comment:
Also I would move this query to the `remove_access_token_for_orcid_account` function, so it can take only `old_orcid` and `new_orcid` args.